build.gradle

```diff
@@ -20,18 +20,32 @@ apply plugin: "io.spring.dependency-management"
 
 dependencyManagement {
     imports {
-        mavenBom 'com.amazonaws:aws-java-sdk-bom:1.11.228'
+        mavenBom 'com.amazonaws:aws-java-sdk-bom:1.11.731'
     }
 }
 
+configurations {
+    // for putting Error Prone javac in bootclasspath for running tests
+    // on Java 8 JDKs
+    errorproneJavac
+}
+
 dependencies {
     testCompile group: 'junit', name: 'junit', version: '4.12'
-    testCompile group: 'org.checkerframework', name: 'testlib', version: '2.8.+'
+    testCompile group: 'org.checkerframework', name: 'framework-test', version: '3.1.+'
+
+    compile group: 'org.checkerframework', name: 'checker', version: '3.1.+'
 
-    compile group: 'org.checkerframework', name: 'checker', version: '2.8.+'
+    errorproneJavac "com.google.errorprone:javac:9+181-r4173-1"
 }
 
 task copyDependencies(type: Copy) {
     from configurations.compile
     into 'dependencies'
 }
+
+test {
+    if (!JavaVersion.current().java9Compatible) {
+        jvmArgs "-Xbootclasspath/p:${configurations.errorproneJavac.asPath}"
+    }
+}
```

Comment:
I think it would be cleaner and less error-prone to use the Gradle plugin at https://github.com/kelloggm/checkerframework-gradle-plugin, rather than giving this configuration (and more on lines 27-32 and 39).

Reply:
The plugin cannot be used for checker development right now, unless you want to run another checker on your checker's code. This code is needed to run the tests for this checker.